sno/working_copy/postgis.py

```diff
@@ -51,19 +65,24 @@ def __init__(self, repo, uri):
             raise ValueError("Expecting postgresql://")
 
         url_path = url.path
-        self.schema = None
-        if url.path:
-            path_parts = url.path[1:].split("/")
-            if len(path_parts) > 2:
-                raise ValueError("Expecting postgresql://[host]/dbname/schema")
-            elif len(path_parts) == 2:
-                self.schema = path_parts[1]
-                url_path = f"/{path_parts[0]}"
-            else:
-                self.schema = self._default_schema(repo)
+        path_parts = url_path[1:].split("/", 3) if url_path else []
+        if len(path_parts) != 2:
+            raise ValueError("Expecting postgresql://[host]/dbname/schema")
```

Comment:
As a user, it's normal to provide [options](https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.6) in the url for timeouts, application names, tls, etc

Reply:
we should probably always set `fallback_application_name=sno`, makes it much easier to see what's going on in the DB via the status queries if there isn't a sno-specific DB user.

Reply:
Done